_posts/2020-03-29-all-you-need-is-lambda-1-booleans.md

```diff
@@ -0,0 +1,260 @@
+---
+layout: post
+title:  "All you need is λ, part one: booleans"
+date:   2020-03-29 20:17:26
+categories: ["lambda calculus", "plt"]
+---
+
+# All you need is λ, part one: booleans
+
+Nearly a hundred years ago, Alonzo Church invented computation in the form of the simple, elegant, and utterly inscrutable lambda calculus. Along with Alan Turing, he then proved that anything computable can be computed in the lambda calculus. However, nearly as soon as we had digital computers, we started inventing programming languages, and with them a vast treasure of features, beautiful and terrible, many of which seem very hard to relate to the fundamental nature of computability, let alone the lambda calculus.
+
+While it’s true that anything which can be computed, period, can be computed in the lambda calculus, you might not want to: it’s austere, to say the least, and was not designed with modern sensibilities regarding readability in mind. We developed all those languages and features for a reason! Still, Church demonstrated not just that it was possible to compute anything computable with the lambda calculus, but also _how_ one might do so.
+
+
+## Unconditional λ
+
+The lambda calculus’ austerity is extreme: you don’t even have booleans. All you have are:
```

Comment:
```suggestion
The lambda calculus’s austerity is extreme: you don’t even have booleans. All you have are:
```

Reply:
This is intentional.

Reply:
Strictly speaking, it's incorrect; suffixed `'` plurals only apply if the word itself is plural _and_ ends in an `s`, e.g. it's `bus's`, `mice's`, but `cheerleaders'`.

Reply:
`calculi's`

Reply:
That would work too.